pkg/operator/ceph/object/spec.go

```diff
@@ -406,14 +409,26 @@ func (c *clusterConfig) vaultPrefixRGW() string {
 	switch secretEngine {
 	case kms.VaultKVSecretEngineKey:
 		vaultPrefixPath = path.Join(vaultPrefixPath,
-			c.clusterSpec.Security.KeyManagementService.ConnectionDetails[vault.VaultBackendPathKey])
+			c.clusterSpec.Security.KeyManagementService.ConnectionDetails[vault.VaultBackendPathKey], "/data")
 	case kms.VaultTransitSecretEngineKey:
 		vaultPrefixPath = path.Join(vaultPrefixPath, secretEngine, "/export/encryption-key")
 	}
 
 	return vaultPrefixPath
 }
 
+func (c *clusterConfig) CheckRGWKMS() (bool, error) {
+	secretEngine := c.clusterSpec.Security.KeyManagementService.ConnectionDetails[kms.VaultSecretEngineKey]
+	kvvers := c.clusterSpec.Security.KeyManagementService.ConnectionDetails[vault.VaultBackendKey]
+
+	//RGW only supports v2 of kv-engine
+	if secretEngine == kms.VaultKVSecretEngineKey && kvvers != "v2" {
+		logger.Error("kv version v1 is not supported for RGW")
```

Comment:
You don't need to both log AND return an error, just return the error with the log message.